common/shm.c

```diff
@@ -53,16 +53,21 @@
 #include <sys/sem.h>
 #include <stdlib.h>
 #include "promiscuous.h"
-
 #endif
 
 #include "shm.h"
 #include "JackError.h"
 
 static int GetUID()
 {
-#ifdef WIN32
-    return  _getpid();
+#if defined(WIN32) || defined(__WIN32)
+    #if defined(__HAVE_USERENV_H)
```

Comment:
Where does __HAVE_USERENV_H come from? And did you mean _WIN32 instead of __WIN32?

Reply:
yes, this is wrong win32, I fix

`__HAVE_USERENV_H` is kinda still in progress, I'm trying to get uid without resorting to a posix->win library. It's defiend in waf builder, where it's marked for check (like with cmake check flags).

Reply:
The problem is that while I find the header file for GetUser, I can't seem to find which library I need to link it with, but it's a WIP, will comment here when I have a candidate solution.